be1-go/message/query_test.go

```diff
@@ -0,0 +1,158 @@
+package message
+
+import (
+	"bytes"
+	"encoding/json"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/require"
+	"golang.org/x/xerrors"
+)
+
+var channel = "/root/channel_id"
+
+func createMessage(i int, timestamp Timestamp) (Message, error) {
+	data, err := NewCreateLAOData("test", timestamp, []byte{byte(i)}, []PublicKey{})
+	if err != nil {
+		return Message{}, err
+	}
+
+	msg := Message{
+		MessageID:         []byte{byte(i)},
+		Data:              data,
+		Sender:            []byte{byte(i)},
+		Signature:         []byte{byte(i)},
+		WitnessSignatures: []PublicKeySignaturePair{},
+	}
+
+	return msg, nil
+}
+
+func compareQuery(q1, q2 Query) (bool, error) {
+	m1, m2 := q1.GetMethod(), q2.GetMethod()
+	if m1 != m2 {
+		return false, nil
+	}
+	if q1.GetChannel() != q2.GetChannel() {
+		return false, nil
+	}
+	if m1 != "broadcast" {
+		if q1.GetID() != q2.GetID() {
+			return false, nil
+		}
+	}
+	if m1 == "subscribe" || m1 == "unsubscribe" {
+		return true, nil
+	}
```

Comment:
you could do a switch here on m1 , dealing with broadcast, subscribe, unsubscribe and the default case. 
it might be a tiny bit more readable.